### PR TITLE
feat: apply for  job save  a job, filter jobs 

### PR DESCRIPTION
**Pull request by Daniekeys**
#### PR Checklist

- [ ] `npm run lint` passes

### CODE REVIEW

File: api-endpoints/jobs/jobs.interface.ts [hunk 1]
```diff
@@ -117,6 +119,7 @@ export interface IJobDetailResponse {
   publish: boolean;
   expiry: string;
   external_data: ExternalData;
+  salaryType: string | undefined | any;
```
Comment (sammy-code98) [inline api-endpoints/jobs/jobs.interface.ts:122]: why using undefined and any types here

File: components/Dashboard/candidate/jobs/index.tsx [hunk 2]
```diff
@@ -1,53 +1,88 @@
...
 const options = [
   { value: "chocolate", label: "Chocolate" },
   { value: "strawberry", label: "Strawberry" },
   { value: "vanilla", label: "Vanilla" },
 ];
 
 const location = [
-  { value: "lagos", label: "Lagos" },
-  { value: "aba", label: "Aba" },
+  { value: "London", label: "London" },
```
Comment (sammy-code98) [inline components/Dashboard/candidate/jobs/index.tsx:15]: use the location array added for the recruiter create job

File: components/jobCard/JobApplicationModal.tsx [hunk 3]
```diff
@@ -108,7 +133,16 @@ const JobApplicationModal: React.FC<ApplyModalProps> = ({
               <span className="text-grey_3 text-lg">Salary</span>
             </div>
             <p className="text-primary_green  font-bold text-lg ml-8 mt-1">
-              {salary}
+              {
+                salaryType && (<span>
+                  
+                  {
+                    salaryType === "range" ? `${salary?.min} - ${salary?.max}` : ` ${salary?.amount}`
```
Comment (sammy-code98) [inline components/jobCard/JobApplicationModal.tsx:140]: please format this line properly

File: components/jobCard/applyJobCard.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    }, { enabled: false })
+
+    const handleModalOpen = () => {
+        sessionStorage.setItem("jobId", _id as string)
+        setOpen(!open)
+        refetch()
+    }
+
+
+    const handleNothing = () => {
```
Comment (sammy-code98) [inline components/jobCard/applyJobCard.tsx:76]: remove this funcyion if it doesnt do anything

**Review by sammy-code98 — CHANGES_REQUESTED** [review]

**Review by chimdie — APPROVED** [review]
@sammy-code98 @copy-ninja1 @Daniekeys  LGTM

File: api-endpoints/jobs/jobs.interface.ts (hunk 1 above)
Comment (Daniekeys) [reply api-endpoints/jobs/jobs.interface.ts:122]: fixed

File: components/jobCard/JobApplicationModal.tsx (hunk 3 above)
Comment (Daniekeys) [reply components/jobCard/JobApplicationModal.tsx:140]: done

**Review by sammy-code98 — APPROVED** [review]

**Pull request merged**